src/main/kotlin/inc/evil/d17map/security/jwt/JWTService.kt

```diff
@@ -0,0 +1,65 @@
+package inc.evil.d17map.security.jwt
+
+
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jwts
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.stereotype.Service
+import java.security.NoSuchAlgorithmException
+import java.util.*
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+
+@Service
+class JWTService {
+
+    private val sec: SecretKey = try {
+        val keyGenerator = KeyGenerator.getInstance("HmacSHA256")
+        keyGenerator.generateKey()
+    } catch (e: NoSuchAlgorithmException) {
+        throw RuntimeException(e)
+    }
+
+    fun generateToken(username: String): String {
+        val claims: Map<String, Any> = HashMap()
+        return Jwts.builder()
+            .claims()
+            .add(claims)
+            .subject(username)
+            .issuedAt(Date(System.currentTimeMillis()))
+            .expiration(Date(System.currentTimeMillis() + 1000 * 60 * 60 * 24)) // 24-hour expiration
```

Comment:
Może warto tutaj stworzyć jakąś stałą, żeby uniknąć podawania w taki sposób liczb?

`companion object {
    const val EXPIRATION_TIME: Long = 1000 * 60 * 60 * 24 // 24 hours
}
`

Reply:
Potem można to zapisać trochę bardziej czytelnie  

````
.expiration(Date(System.currentTimeMillis() + EXPIRATION_TIME))

Reply:
Wprowadziłem obsługę propertiesów zamiast stałych w kodzie, aby `TTL` tokenu był modyfikowalny w plikach konfiguracyjnych.